narwhals/_expression_parsing.py

```diff
@@ -338,11 +347,46 @@ def extract_compliant(
     return other
 
 
+def arg_aggregates(arg: IntoExpr | Any) -> bool:
+    from narwhals.expr import Expr
+    from narwhals.series import Series
+
+    if isinstance(arg, Expr):
+        return arg._metadata["aggregates"]
+    if isinstance(arg, Series):
+        return arg.len() == 1
+    if isinstance(arg, str):  # noqa: SIM103
+        # Column name, e.g. 'a', gets treated as `nw.col('a')`,
+        # which doesn't aggregate.
+        return False
+    # Scalar
+    return True
+
+
+def arg_is_order_dependent(arg: IntoExpr | Any) -> bool:
+    from narwhals.expr import Expr
+
+    if isinstance(arg, Expr):
+        return arg._metadata["is_order_dependent"]
+    if isinstance(arg, str):
+        # Column name, e.g. 'a', gets treated as `nw.col('a')`,
```

Comment:
sigh...this isn't even always true. darn. we may need to parse column names as expressions a bit earlier